src/prog_models/prognostics_model.py

```diff
@@ -342,7 +402,59 @@ def next_state(self, x, u, dt) -> dict:
         
         # Note: Default is to use the dx method (continuous model) - overwrite next_state for continuous
         dx = self.dx(x, u)
-        return {key: x[key] + dx[key]*dt for key in x.keys()}
+        return {key: x[key] + dx[key]*dt for key in dx.keys()}
+
+    def __next_state(self, x, u, dt) -> dict:
+        """
+        State transition equation: Calls next_state(), calculating the next state, and then adds noise
+
+        Parameters
+        ----------
+        x : dict
+            state, with keys defined by model.states \n
+            e.g., x = {'abc': 332.1, 'def': 221.003} given states = ['abc', 'def']
+        u : dict
+            Inputs, with keys defined by model.inputs \n
+            e.g., u = {'i':3.2} given inputs = ['i']
+        dt : number
+            Timestep size in seconds (≥ 0) \n
+            e.g., dt = 0.1
+
+        Returns
+        -------
+        x : dict
+            Next state, with keys defined by model.states
+            e.g., x = {'abc': 332.1, 'def': 221.003} given states = ['abc', 'def']
+
+        Example
+        -------
+        | m = PrognosticsModel() # Replace with specific model being simulated
+        | u = {'u1': 3.2}
+        | z = {'z1': 2.2}
+        | x = m.initialize(u, z) # Initialize first state
+        | x = m.__next_state(x, u, 0.1) # Returns state, with noise, at 3.1 seconds given input u
+        
+        See Also
+        --------
+        next_state
+
+        Note
+        ----
+        A model should not overwrite '__next_state'
+        A model should overwrite either `next_state` or `dx`. Override `dx` for continuous models, and `next_state` for discrete, where the behavior cannot be described by the first derivative.
+        """
+        
+        # Calculate next state and add process noise
+        next_state = self.apply_process_noise(self.next_state(x, u, dt))
+
+        # Check if state is within bounds
+        for (key, limit) in self.state_limits.items():
```

Comment:
This may not be feasible for this release, but it may make more sense to sample directly from a truncated distribution (that fits within the limits) instead of setting the state to the exact value of the limit. That way the limit itself is never reached by the state. This could be done with a modification on apply_process_noise.